src/closure.md

```diff
@@ -0,0 +1,113 @@
+# Closure Expansion in rustc
+
+Let's start with an example
+
+### Example 1
+```rust
```

Comment:
A little introduction here too perhaps. Maybe

---

To start, let's take a look at how the closure in the following example is desugared: